tests/loaders/test_youtube_video.py

```diff
@@ -1,53 +1,64 @@
 import hashlib
-from unittest.mock import MagicMock, Mock, patch
+import json
+from unittest.mock import patch
 
 import pytest
+import yt_dlp
+from youtube_transcript_api import YouTubeTranscriptApi
 
-from embedchain.loaders.youtube_video import YoutubeVideoLoader
-
+from embedchain.loaders.youtube_video import YoutubeVideoLoader  # Adjust the import according to your module structure
 
 @pytest.fixture
 def youtube_video_loader():
     return YoutubeVideoLoader()
 
-
 def test_load_data(youtube_video_loader):
     video_url = "https://www.youtube.com/watch?v=VIDEO_ID"
-    mock_loader = Mock()
-    mock_page_content = "This is a YouTube video content."
-    mock_loader.load.return_value = [
-        MagicMock(
-            page_content=mock_page_content,
-            metadata={"url": video_url, "title": "Test Video"},
-        )
-    ]
-
+    mock_result = {
+        'title': 'Test Video',
+        'uploader': 'Uploader Name',
+        'upload_date': '20200101',
+        'duration': 3600,
+        'view_count': 1000,
+        'like_count': 100,
+        'dislike_count': 10,
+        'average_rating': 4.5,
+        'categories': ['Education'],
+        'tags': ['tag1', 'tag2'],
+        'id': 'VIDEO_ID'
+    }
     mock_transcript = [{"text": "sample text", "start": 0.0, "duration": 5.0}]
+    serialized_transcript = json.dumps(mock_transcript, ensure_ascii=False)  # Serialize the transcript for the test
+    content = ' '.join([item['text'] for item in mock_transcript])
 
-    with patch("embedchain.loaders.youtube_video.YoutubeLoader.from_youtube_url", return_value=mock_loader), patch(
-        "embedchain.loaders.youtube_video.YouTubeTranscriptApi.get_transcript", return_value=mock_transcript
-    ):
+    with patch.object(yt_dlp.YoutubeDL, 'extract_info', return_value=mock_result), \
+         patch.object(YouTubeTranscriptApi, 'get_transcript', return_value=mock_transcript):
```

Comment:
Same here as well, can you please use `embedchain.loaders.youtube_video. YoutubeVideoLoader. YouTubeTranscriptApi`